### DISCUSSION TITLE
[Question] Is their any available palette component similar to the "Theme controls" on material-web.dev?

### DISCUSSION THREAD

**Discussion by creamIcec** [material-components/material-web#5555] [Q&A] [answered]
Just a quick question may be not relative to this repo, is there an available "Theme controls" component which appeared on 
https://material-web.dev/ at right top corner?(The "palette" icon) 
Currently I'm copying color schema generated by it, and I find it so useful for changing color and copying values.
Is this a component I can use on my website for user to change their preferred color? 
If not, where can I find algorithm to generate hex values behind the scene, for myself to implement this functionality?
Thank you in advance.

**Comment by asyncliz** [comment, marked as answer]
You can copy and/or reference the [catalog's theme changer component here](https://github.com/material-components/material-web/blob/b46361ea7492f3feef8209681d163e0f2121bf52/catalog/src/components/theme-changer.ts)! It uses [some utility functions](https://github.com/material-components/material-web/blob/b46361ea7492f3feef8209681d163e0f2121bf52/catalog/src/utils/material-color-helpers.ts) that build on top of [`@material/material-color-utilities`](https://www.npmjs.com/package/@material/material-color-utilities), which is the algorithm that generates Material 3 color palettes.

**Reply by creamIcec** [reply]
Thank you for quick reply! I will soon check out this.
